### DISCUSSION TITLE
what do we do with las_check results?

### DISCUSSION THREAD

**Discussion by adrienmichez** [r-lidar/lidR#487] [Q&A]
Hi,

It would be nice to have some kind of guidelines about how to fix - or at least knowing the potential issues associated with - the results of las_check ?

Maybe a section in the lidr book?

Just a suggestion... Thanks again for your work :)

Adrien

**Comment by Jean-Romain** [comment]
No way to do that, Each problem is data specific and a similar problems can have different unknown causes. And there is an infinite number ways to be invalid. You must figure out why your data provider provided you invalid data. I recently found an invalid dataset where the number of return was set to 0 to pulses with more than 7 returns because it is not possible to put more than 7 in this attribute. The LAS specification states that pulses have up to 5 returns. Only human investigation could have guessed it and there are millions of other (bad) reasons to find zeros in this attribute. Tell me what your problem is and I'll see if I can help.

**Reply by adrienmichez** [reply]
Thanks for your reply

I posted some more details on https://gis.stackexchange.com/questions/413807/fix-point-cloud-lidr-normalize-height-fails
